libs/forecasting/src/lib/ForecastingDashboard/Controls/TransformationCreationDialog.tsx

```diff
@@ -191,6 +181,38 @@ export class TransformationCreationDialog extends React.Component<
       </Dialog>
     );
   }
+  getTransformationValueErrorMessage() {
+    if (
+      this.state.transformationOperation &&
+      this.state.transformationFeature
+    ) {
+      const featureMeta =
+        this.context.jointDataset.metaDict[
```

Comment:
you index into this.context.jointDataset a few times; just want to make sure that this will always be defined

Reply:
JointDataset isn't optional, but the context may not be defined yet (in the first render before the context is set up. I'm adding `&& this.context` to the `if` right above. LMK if that matches your expectations here.

Reply:
Taking this offline for a separate discussion, as this is something affecting hundreds of places within the codebase.